crates/autopilot/src/database/onchain_order_events.rs

```diff
@@ -394,8 +397,14 @@ fn convert_onchain_order_placement(
         sell_token_balance: sell_token_source_into(order_data.sell_token_balance),
         buy_token_balance: buy_token_destination_into(order_data.buy_token_balance),
         full_fee_amount: u256_to_big_decimal(&full_fee_amount),
-        is_liquidity_order,
         cancellation_timestamp: None,
+        // TODO #643: To properly determine the class, we have to check the liquidity owners just
+        // like we do during order creation.
+        class: if is_liquidity_order {
+            OrderClass::Liquidity
+        } else {
+            OrderClass::Ordinary
+        },
```

Comment:
Actually not sure how it should work for onchain orders. The comment above, where is_liquidity_owner is defined gives a different answer. I wonder if this is an oversight when @josojo wrote this code recently. The question is, if an onchain order is out of price then is it a liquidity order even the owner is not in the liquidity_owners list? Or is it a limit order?
I agree with your new comment that it should work the same way as what you changed in order_validation. Just tagging Alex for visibility.